data/virt_autotest/guest_console_monitor.sh

```diff
@@ -0,0 +1,284 @@
+    exit 1
+}
+
+function start_monitor()
+{
+        #start monitor for guest console
+        get_vm_guestnames_state_running=`virsh list --state-running | grep -E "${vm_guestnames_types}" | awk '{print $2}'`
+        vm_guestnames_state_running_array=$(echo -e ${get_vm_guestnames_state_running})  
+
+        while :
+        do
+		    if [ -f ${MONITOR_LOCKED_FILE} ];then
+                #Check with LOCK-UP status
+                cat ${MONITOR_LOCKED_FILE} | grep "LOCKED"                
+                if [ $? -eq 0 ];then
+                    #KEEP LOCK-UP status
+                    #To let just only one virsh console with one vm guest as backgroup PID
+                     
+                    for i in `seq 12`;do
+			            echo -e "Please be patient. Will attempt check shortly.\n"
+		                sleep 10s
+			            echo -e "$((120 - $i * 10)) seconds remaining.\n"
+			        done
+
+                    sleep_time=180
+                    sleep_count=0
+                    failed_count=0
+                    unset vmguest_check_failed
+                    declare -a vmguest_check_failed=""
+                    while [ ${sleep_count} -le ${sleep_time} ];do
+                        echo -e "Try $((${sleep_count}/10 + 1)) time(s) check\n"
+                        vmguest_index=0
+
+                        #Check each vm guest and store checked result.
+                        for vmguest in ${vm_guestnames_array[@]};do
+                            #ping -c 1 ${vmguest}
+                            virsh list --state-running | grep ${vmguest}
+                            vmguest_check_failed[${vmguest_index}]=`echo $?`
+                            vmguest_index=$((${vmguest_index} + 1))
+                        done
+                        failed_count=0
+                        vmguest_index=0
+                    
+                        #Iterate vmguest_check_failed to check whether ping to each vm guest succeeded.
+                        for vmguest in ${vm_guestnames_array[@]};do
+                            if [[ ${vmguest_check_failed[${vmguest_index}]} -ne 0 ]];then
+                                echo -e "Connection to ${vmguest} failed !\n"
+                                failed_count=$((${failed_count} + 1))
+                            else
+                                echo -e "Connection to ${vmguest} succeeded !\n"
+                            fi
+                            vmguest_index=$((${vmguest_index} + 1))
+                        done
+                    
+                        #If ping to all vm guests succeeded, terminate while loop and proceed. 
+                        if [[ ${failed_count} -eq 0 ]];then
+                            echo -e "Connection to all vm guests successfully.\n"
+                            break
+                        fi
+                        sleep_count=$((${sleep_count} + 10))
+                        sleep 10s
+                    done			        
```

Comment:
Can we just remove this wait and sleep loops ?

Reply:
remove this wait and sleep loops